### PR TITLE
[3/N] Apply py39 ruff fixes

### PR DESCRIPTION
**Pull request by cyyever**
Fixes #ISSUE_NUMBER

cc @EikanWang @jgong5 @wenzhe-nrv @sanchitintel

### CODE REVIEW

File: tools/autograd/gen_annotated_fn_args.py [hunk 1]
```diff
@@ -39,6 +39,8 @@
 
 
 if TYPE_CHECKING:
+    from collections.abc import Sequence
```
Comment (ezyang) [inline tools/autograd/gen_annotated_fn_args.py:42]: To be honest, it's really unclear to me why this is an improvement

**Review by ezyang — APPROVED** [review]

File: tools/autograd/gen_annotated_fn_args.py (hunk 1 above)
Comment (cyyever) [reply tools/autograd/gen_annotated_fn_args.py:42]: I also don't know why python introduced this flag...

**Pull request closed without merging**